src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1022,19 +1026,6 @@ ExecuteTaskListExtended(ExecutionParams *executionParams)
 			defaultTupleDest, &executionParams->xactProperties,
 			executionParams->jobIdList, executionParams->localExecutionSupported);
 
-	/*
-	 * If current transaction accessed local placements and task list includes
-	 * tasks that should be executed locally (accessing any of the local placements),
-	 * then we should error out as it would cause inconsistencies across the
-	 * remote connection and local execution.
-	 */
-	List *remoteTaskList = execution->remoteTaskList;
-	if (GetCurrentLocalExecutionStatus() == LOCAL_EXECUTION_REQUIRED &&
-		AnyTaskAccessesLocalNode(remoteTaskList))
-	{
-		ErrorIfTransactionAccessedPlacementsLocally();
-	}
```

Comment:
instead, now we call EnsureCompatibleLocalExecutionState in StartDistributedExecution

Reply:
why do we need to change this place as we already handle the problematic case in `CanFailoverPlacementExecutionToLocalExecution `?

Reply:
definitely true, let me fix that quickly

Reply:
even more, took that change into another pr https://github.com/citusdata/citus/pull/5640